### PR TITLE
fix(course): fix dropdown search input behavior on CommandInput

### LINKED ISSUE #375: 手機無法使用進階查詢課程的功能
點擊進階選課的欄位時，像是想查詢二專選課有哪些時，會被鍵盤擋住無法往下拉，造成頁面跳轉
聯絡方式（信箱）921105allen@gmail.com

### LINKED ISSUE #395: 課程進階搜尋 - 手機界面不要預設開鍵盤
題外話：團隊們要不要在問題回報可以附影片圖片？

設備：iphone 8 plus 但我想沒返回鍵的手機都有這問題
如題，如果今天我想篩選課程，點漏斗標籤後會進到可以選標籤系所的小彈窗
但是要選擇系所時，會預設我使用鍵盤打字搜尋，原本下拉的選單變成上展突出畫面，怎麼拉也拉不下來，最上面的選項都看不到。想點空白處收起鍵盤用滑的來選，系統又會判定我選完了收起選單

希望點擊選單不要預設打字搜尋，然後改掉要打字搜尋選單就會往上展
不然想找個課很麻煩，尤其這過程中還很常卡卡沒回應或當機！！
拿朋友的i15試過了一樣所以應該不是手機太爛的問題

### PR DESCRIPTION
**Pull request by pedestrianlove**
This fix introduces the following new behaviors as a workaround:
- The input mode for drop down menus will now default to `none`.
- If the menu is opened, and user clicks on the `CommandInput`, it'll change its input mode back to `text`.
- If the menu is opened, and user scrolls the `ScrollArea`, it'll change its input mode back to `none`.

---
- For more info on `inputmode`, please refer to MDN: 
  - https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode
  - TL;DR 
    It defaults to `text`, and `none` will disable keyboard popup on mobile devices.
-  This should fix #375 and fix #395

### CODE REVIEW

**Comment by ImJustChew** [comment]
😍 Finally someone's submitting a PR

Looks good! This is my first time knowing there's an attribute for this, very cool. 

However, disableTyping is not triggered when the dropdown is closed. You can recreate this by opening the dropdown search twice, and the 2nd time will cause the virtual keyboard to appear again. 

Other than that, thanks!

**Comment by pedestrianlove** [comment]
> 😍 Finally someone's submitting a PR
> 
> Looks good! This is my first time knowing there's an attribute for this, very cool.
> 
> However, disableTyping is not triggered when the dropdown is closed. You can recreate this by opening the dropdown search twice, and the 2nd time will cause the virtual keyboard to appear again.
> 
> Other than that, thanks!

OK, I've added the `setEnableTyping` callback to the `openChange` function to ensure that `typable` is always initialized to `false` when the menu is opened. 

I also noticed that using two separate functions for such simple state management might be a bit overkill, therefore I've inlined the logic directly into the component.  This should make the code easier to read.

**Review by ImJustChew — COMMENTED** [review]
Looks good~

**Review by ImJustChew — APPROVED** [review]
